webapi/Storage/CosmosDbContext.cs

```diff
@@ -100,7 +100,7 @@ public async Task UpsertAsync(T entity)
             throw new ArgumentOutOfRangeException(nameof(entity.Id), "Entity Id cannot be null or empty.");
         }
 
-        await this._container.UpsertItemAsync(entity);
+        await this._container.UpsertItemAsync(entity, new PartitionKey(entity.Partition));
```

Comment:
I thought the partition key was automatically extracted from the entity in this case.  It certainly doesn't hurt to be explicit.

Reply:
Yeah it is, but this is mostly to align our sanity check with the new partitionKeys, i.e, before update would work with Cosmos containers configured before the new key requirements because it was auto-extracting the default partition key rather than throwing an error when trying to reference the new key value